app/src/main/java/movierating/services/UserServiceImpl1.java

```diff
@@ -12,7 +13,9 @@ public UserServiceImpl1(UserRepository userRepository) {
     }
 
     @Override
-    public void addUser(User user) {
+    public void addUser(User user) throws AddUserException {
```

Comment:
I think better name would be `UserAlreadyExistsException`. I think so because an exception is supposed to tell you the reason why it is an errored state.

Reply:
Changed the name of this exception.